validate.go

```diff
@@ -44,11 +44,32 @@ func fieldValidators(model reflect.Value, opt ...Option) (map[string]validator,
 
 	fValidators := make(map[string]validator)
 	for i := 0; i < m.NumField(); i++ {
-		if slices.Contains(opts.withIgnoredFields, m.Type().Field(i).Name) {
+		field := m.Type().Field(i)
+		if slices.Contains(opts.withIgnoredFields, field.Name) {
 			continue
 		}
 
-		fName := strings.ToLower(m.Type().Field(i).Name)
+		// Determine field name to use for mapping
+		// First check if we have a tag specified and if the field has that tag
+		fName := ""
```

Comment:
Same bits, just a suggested implementation all within a switch and a bit more explicit.
```
	var fName string
    switch {
    case opts.withColumnFieldTag != "": 
        tagValue := field.Tag.Get(opts.withColumnFieldTag)
        if tagValue != "" { 
            parts := strings.SplitN(tagValue, ",", 2)
            fName = parts[0]
        }
        if fName == "" {
            return nil, fmt.Errorf("%s: field %q has an invalid tag %q: %w", op, field.Name, opts.withColumnFieldTag, ErrInvalidParameter)
        }
    default:
        fName = strings.ToLower(field.Name)
    }
```

Reply:
This implies you want no safe fallback to field name?

Reply:
I'm suggesting that a bad tag should be an error and not unexpectedly fall back
to the field name.  The caller explicitly stated that they want
withColumFieldTag and we found a tag.  If that tag has an empty string for the
name, I think that's an error.